### PR TITLE
Add NightBot support and improve performance

### PR DESCRIPTION
**Pull request by craftwar**
I don't maintain this branch anymore. I rewrite it as my extension.
Do you mind share information about audio url? (doc or discuss thread is preferred) That's the only part I don't understand. I rewrite anything else except that. I can make better extension if I understand it.

### CODE REVIEW

File: js/youtube_audio.js [hunk 1]
```diff
@@ -1,35 +1,40 @@
-function makeSetAudioURL(videoElement, url) {
-    function setAudioURL() {
-        if (videoElement.src  != url) {
-            videoElement.pause();
-            videoElement.src = url;
-            videoElement.currentTime = 0;
-            videoElement.play();
-        }
+var oldUrl;
```
Comment (animeshkundu) [inline js/youtube_audio.js]: Not sure, why do you require this.

**Comment by animeshkundu** [comment]
Would love to get the above changes in. 

About the audio url, would you mind being a bit more specific. Would be happy to answer them.

File: js/youtube_audio.js (hunk 1 above)
Comment (craftwar) [reply js/youtube_audio.js]: I planned to reset video without reloading. But I give up, don't think it's worth spending much time.
You can delete whatever you don't want.

Comment (animeshkundu) [reply js/youtube_audio.js]: Okay, I will go ahead an merge it after removing this.

**Pull request merged**